### ISSUE TITLE
aws_batch_job_definition timeout parameter needs to trigger resource recreation

### ISSUE THREAD

**Issue by hectcastro** [terraform-providers/terraform-provider-aws#4696]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```bash
Terraform v0.11.7
```

### Affected Resource(s)

* `aws_batch_job_definition`

### Terraform Configuration Files

```hcl
resource "aws_batch_job_definition" "ingest_scene" {
  name = "jobIngestScene"
  type = "container"

  container_properties = "${data.template_file.ingest_scene_job_definition.rendered}"

  parameters {
    sceneId = " "
  }

  retry_strategy {
    attempts = 3
  }

  timeout {
    attempt_duration_seconds = 7200
  }
}
```

### Expected Behavior

When `timeout.attempt_duration_seconds` is updated, the corresponding task definition should reflect the change.

### Actual Behavior

When `timeout.attempt_duration_seconds` is updated, Terraform attempts to update the attribute, but the resource does not support updates.

```
* aws_batch_job_definition.ingest_scene: 1 error(s) occurred:

* aws_batch_job_definition.ingest_scene: doesn't support update
```

### Steps to Reproduce

1. Modify the existing timeout parameter of a `aws_batch_job_definition`
2. `terraform apply`
3. See the error message above

**Comment by bflad** [comment]
Thanks for reporting this! Looks like the top level `timeout` attribute was marked `ForceNew: true` but not the underlying `attempt_duration_seconds` attribute:

https://github.com/terraform-providers/terraform-provider-aws/blob/333abdf5cc9956f777ffd08e904fff86d9690e38/aws/resource_aws_batch_job_definition.go#L60-L74

If you would like to submit the one-line fix, I'll gladly turbo-merge it and we can get it released. 👍

**Comment by bflad** [comment]
This has been released in [version 1.21.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1210-may-31-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
